src/Credentials.js

```diff
@@ -60,7 +65,7 @@ export default class Credentials {
       if(this.settings.address) {
         if(payload.req) {
           return verifyJWT(this.settings, payload.req).then((challenge) => {
-            if(challenge.payload.iss === this.settings.address) {
+            if(challenge.payload.iss === this.settings.address && challenge.payload.type === 'shareReq') {
```

Comment:
@pelle just to make sure that the token being received is for a share request, before processing it